src/Components/Endpoints/src/DependencyInjection/RazorComponentOptions.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.Endpoints.FormMapping;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Provides options for configuring server-side rendering of Razor Components.
+/// </summary>
+public class RazorComponentOptions
+{
+    internal readonly FormDataMapperOptions _formMappingOptions = new();
+
+    /// <summary>
+    /// Gets or sets the maximum number of elements allowed in a form collection.
+    /// </summary>
+    public int MaxFormMappingCollectionSize
+    {
+        get => _formMappingOptions.MaxCollectionSize;
+        set
+        {
+            ArgumentOutOfRangeException.ThrowIfNegativeOrZero(value);
+            _formMappingOptions.MaxCollectionSize = value;
+        }
+    }
```

Comment:
Maybe this is a question for API review, but would we consider having a top-level `FormOptions` property rather than promoting form data mapper options to `RazorComponentOptions` properties?

Reply:
This is deliberate since we don't want to expose any public API for form data mapping, as it's shared with minimal APIs. The plan here is that in .NET 9.0 we'll move things to a common location, and at that point, we forward those properties to the public options type.